### PR TITLE
Document node.setonerror()

### LINKED ISSUE #3381: Document node.setonerror()
### Expected behavior
Documentation should contain node.setonerror()

### Actual behavior
Not contained

### Test code
See https://nodemcu.readthedocs.io/en/dev/search.html?q=setonerror

See https://gist.github.com/TerryE/646407100785fb00c2cbddbb0e8a2b61#panic-handling for more Information

### PR DESCRIPTION
**Pull request by marcelstoer**
Fixes #3381.

### CODE REVIEW

**Comment by nwf** [comment]
Probably worth _strongly encouraging_ people to ensure that the callback ends with a restart.  Something has gone quite wrong and it's probably not safe to just wait for the next event (e.g., timer tick) and hope everything works out.

**Comment by marcelstoer** [comment]
Very valid point, thanks! I amended the description and the snippet.

File: docs/modules/node.md [hunk 1]
```diff
@@ -481,6 +480,32 @@ target CPU frequency (number)
 node.setcpufreq(node.CPU80MHZ)
 ```
 
+## node.setonerror()
+
+Use this to override the default "always restart" action if wanted; for example to write an error to a logfile or to a network syslog before restarting.
```
Comment (HHHartmann) [inline docs/modules/node.md:485]:
```suggestion
Overrides the default crash handling which always restarts the system; Can be used to write an error to a logfile or to secure connected hardware before restarting.
```

Should be in the style of the rest of the docs

**Review by nwf — APPROVED** [review]

**Comment by HHHartmann** [comment]
I wonder if we should document how the default reset works with its 2 seconds wait before reboot.
Entirely not sure if this is rhe right place though.

**Comment by marcelstoer** [comment]
Rather document too much than too little. Better to have it in the wrong place than nowhere at all.

If you propose a few sentences and could also add them to the FAQ and link to it from this function.

**Comment by marcelstoer** [comment]
Gregor, do you still want to amend this or can we merge as-is?

**Pull request merged**